opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/RuntimeEnvironment.java

```diff
@@ -775,15 +807,15 @@ public void setDefaultProjects(Set<Project> defaultProject) {
      * @return the default projects (may be null if not specified)
      */
     public Set<Project> getDefaultProjects() {
-        return threadConfig.get().getDefaultProjects();
+        return (Set<Project>)getConfigurationValue("defaultProjects");
```

Comment:
if you modify the returned set from multiple threads you have a problem here

Reply:
right, the returned set needs to be immutable.